### DISCUSSION TITLE
Support invisible reCaptcha, maybe also for login screen

### DISCUSSION THREAD

**Discussion by jbman** [keycloak/keycloak#23562] [Ideas]
While testing the current reCaptcha integration, I found that it did work with a visible Google reCaptcha v2 ("I'm not a robot" checkbox).
Google provides v3 which allows verification without user interaction and provides a score. Also the v2 invisible might be a more convenient option for the user.
https://www.google.com/recaptcha/about/ 
https://developers.google.com/recaptcha/docs/faq

Is it planned to or any interest from other parties to support reCaptcha v3 or at least invisible captcha at Keycloak?

What about integration at the login screen?
It could to prevent automated login attempts without the need for a strict brute force prevention. This would show challenges to solve instead of complete user locking.

**Comment by cesarcoruna** [comment]
We have recently integrated Recaptcha v3 into our login page. The problem seems to be that Recaptcha v3 needs to be used on several previous pages in order to make a correct decision. When used only in the login page the score we get is always the maximum (0.9) so it is totally useless. 

In our case using Recaptcha v3 in previous pages is complicated because they are applications developed by other entities so we are thinking about the possibility of removing it from the login page because it does not contribute anything.
